<launch>
  <!-- 선택 파라미터 -->
  <arg name="vesc_port" default="/dev/ttyVesc"/>
  <arg name="use_vesc" default="true"/>

  <!-- VESC 드라이버 (필요 시) -->
  <group if="$(arg use_vesc)">
    <include file="$(find vesc_driver)/launch/vesc_driver_node.launch">
      <arg name="port" value="$(arg vesc_port)"/>
    </include>
  </group>

  <!-- 카메라 + 이미지 보정 -->
  <include file="$(find main)/launch/usb_cam_headless.launch"/>
  <node pkg="image_proc" type="image_proc" name="image_proc" ns="/usb_cam" output="screen"/>

  <!-- 단순 차선 추종 노드 -->
  <node name="lane_follower" type="main_run.py" pkg="main" output="screen"/>
</launch>
